File: Patterns/Abstract Factory Pattern/Cpp/PizzaStore/ChicagoPizzaStore.cpp
#include "../Pizza/Pizza.cpp"

#include "../Pizza/Ingredient/PizzaIngredientFactory.cpp"
#include "../Pizza/Ingredient/ChicagoPizzaIngredientFactory.cpp"

#include "../Pizza/CheesePizza.cpp"
#include "../Pizza/ClamPizza.cpp"
#include "../Pizza/PepperoniPizza.cpp"
#include "../Pizza/VeggiePizza.cpp"

#include "PizzaStore.cpp"

class ChicagoPizzaStore : public PizzaStore {
    Pizza* pizza = nullptr;
    PizzaIngredientFactory* ingredientFactory = new ChicagoPizzaIngredientFactory();

    Pizza* createPizza(string type) override {
        if (type == "cheese") {
            pizza = new CheesePizza(ingredientFactory);
            pizza->setName("시카고 스타일 치즈 피자\n");
        }
        else if (type == "clam") {
            pizza = new ClamPizza(ingredientFactory);
            pizza->setName("시카고 스타일 조개 피자\n");
        }
        else if (type == "pepperoni") {
            pizza = new PepperoniPizza(ingredientFactory);
            pizza->setName("시카고 스타일 페페로니 피자\n");
        }
        else if (type == "veggie") {
            pizza = new VeggiePizza(ingredientFactory);
            pizza->setName("시카고 스타일 채소 피자\n");
        }
        
        return pizza;
    }
};

File: Patterns/Abstract Factory Pattern/CSharp/PizzaStore/ChicagoPizzaStore.cs
using Ingredient = Pizza.Ingredient;

namespace PizzaStore
{
    public class ChicagoPizzaStore : PizzaStore
    {
        private Pizza.Pizza? pizza;
        private Ingredient.PizzaIngredientFactory ingredientFactory;

        public ChicagoPizzaStore() : base() 
        {
            pizza = null;
            ingredientFactory = new Ingredient.ChicagoPizzaIngredientFactory();
        }

        protected override Pizza.Pizza? CreatePizza(string type) 
        {
            switch (type)
            {
                case "cheese" : 
                    pizza = new Pizza.CheesePizza(ingredientFactory);
                    pizza.Name = "시카고 스타일 치즈 피자";
                    break;
                case "clam" : 
                    pizza = new Pizza.ClamPizza(ingredientFactory);
                    pizza.Name = "시카고 스타일 조개 피자";
                    break;
                case "pepperoni" : 
                    pizza = new Pizza.PepperoniPizza(ingredientFactory);
                    pizza.Name = "시카고 스타일 페페로니 피자";
                    break;
                case "veggie" : 
                    pizza = new Pizza.VeggiePizza(ingredientFactory);
                    pizza.Name = "시카고 스타일 채소 피자";
                    break;
            }

            return pizza;
        }
    }
}